### PR TITLE
[labs/virtualizer] Fix 3243

### LINKED ISSUE #3243: [lab/virtualizer] DOM update doesn't successfully complete under some circumstances
### Which package(s) are affected?

Virtualizer (@lit-labs/virtualizer)

### Description

We have a scenario in our app where the LitVirtualizer doesn't render horizontal flow correctly on the first load in the Safari browser. A subsequent update to the items list seems to fix the issue. 

- This only reproduces within our app and Safari 
- I'm unable to reproduce this when isolating this into its own [sandbox](https://codesandbox.io/s/litelement-typescript-litvirtualizer-sxjsww?file=/src/x-row-scroller.ts)

Our code specified this:
```
<x-virtualizer
    style="min-height: 50px"
    .layout=${flow({ direction: "horizontal" })}
    scroller
    .items=${this.items}
    .renderItem=${this._renderItemProxy}
></x-virtualizer>
```

When inspecting the DOM on Safari during the first load I noticed the elements inside the LitVirtualizer did not the transform style it needs to layout out the horizontally:
```
<x-virtualizer style="min-height: 50px; display: block; position: relative; contain: strict; overflow: auto;" scroller="" dir="ltr">
<div virtualizer-sizer="" style="position: absolute; margin: -2px 0px 0px; padding: 0px; visibility: hidden; font-size: 2px; transform: translate(1000px, 0px);">&nbsp;</div>
            <div class="proxy-item-ltr " id="idx-0">
            </div>
            <div class="proxy-item-ltr " id="idx-1">
            </div>
            <div class="proxy-item-ltr " id="idx-2">
            </div>     
</x-virtualizer> 
```

When it's working correctly we can see that in-style is applied to these items:
```
<x-virtualizer style="min-height: 50px; display: block; position: relative; contain: strict; overflow: auto;" scroller="" dir="ltr">
    <div class="proxy-item-ltr " id="idx-0" style="position: absolute; box-sizing: border-box; transform: translate(0px, 0px); left: 0px;">      
    </div>
    <div class="proxy-item-ltr " id="idx-1" style="position: absolute; box-sizing: border-box; transform: translate(216px, 0px); left: 0px;">     
    </div>
    <div class="proxy-item-ltr " id="idx-2" style="position: absolute; box-sizing: border-box; transform: translate(432px, 0px); left: 0px;            
    </div>
</x-virtualizer>
```

### Reproduction

Initial of the our app (unfortunately can't share to public yet)

### Workaround

In the component initialize the `items` array with an initial value and on `firstUpdated()` life cycle method update the `items` array again with a different sets of value

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

"@lit-labs/virtualizer": "0.7.1"

### Browser/OS/Node environment

Browser: Version 15.6 (17613.3.9.1.5)

### PR DESCRIPTION
**Pull request by graynorton**
Fixes #3243 (and #3498, which was revealed in our automated tests by the fix for #3243)

### CODE REVIEW

**Review by usergenic — CHANGES_REQUESTED** [review]
I had to try and see if this fixed #3496 but it didn't.

File: packages/labs/virtualizer/src/test/layouts/flow.test.ts [hunk 1]
```diff
@@ -194,9 +194,10 @@ describe('flow layout', () => {
       // No change in visible items is expected since item 5 is already visible.
```
[the remark this answers is not in the document]
Comment (graynorton) [reply packages/labs/virtualizer/src/test/layouts/flow.test.ts:194]: Yes—the fact that the visible range didn’t change after our last call to `scrollIntoView()` is taken as evidence that (as expected) no actual scrolling was required, nor occurred. We could actually verify more strongly by checking the precise scroll position before and after, but I left the previous assertion logic intact and just fixed the timing flaw that was allowing the the test to false-positive before the fix for #3243 went in.

File: packages/labs/virtualizer/src/test/scenarios/immediate-items-change.test.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+// use of lit-mobx. Other minimal repro cases are probably possible.
+// The fix we've identified for this issue is a low-level change to the logic
+// for completing the virtualizer update cycle after children have been
+// rendered or re-rendered.
+describe('Successful DOM update on immediate change to items', () => {
+  ignoreBenignErrors(beforeEach, afterEach);
+
+  it('should apply transforms to DOM children', async () => {
+    const example = (await fixture(
+      testingHtml`<my-example></my-example`
```
[the remark this answers is not in the document]
Comment (graynorton) [reply packages/labs/virtualizer/src/test/scenarios/immediate-items-change.test.ts:51]: Thanks for the catch! Will fix.

**Review by usergenic — APPROVED** [review]
LGTM

golf:
`Promise((resolve) => requestAnimationFrame(resolve));`

equivalent?
`Promise(requestAnimationFrame);`

**Pull request merged**